src/tests/ftest/telemetry/engine_events.py

```diff
@@ -0,0 +1,353 @@
+
+        Args:
+            rank_count (int): Rank count.
+            restart_rank (int): Restarted rank.
+            events_last_event_ts_0 (dict): engine_events_last_event_ts values before.
+            events_last_event_ts_1 (dict): engine_events_last_event_ts values after.
+            errors (list): Errors.
+            events_last_event_ts_results (list): Dictionary to store the results, which are printed
+                at the end of the test.
+        """
+        for rank in range(rank_count):
+            result = ""
+            if rank == restart_rank:
+                if events_last_event_ts_0[rank] != events_last_event_ts_1[rank]:
+                    msg = (f"engine_events_last_events_ts value for restarted rank {rank} "
+                           f"changed! Before = {events_last_event_ts_0[rank]}; "
+                           f"After = {events_last_event_ts_1[rank]}")
+                    errors.append(msg)
+                    result = (f"Rank {rank}: {events_last_event_ts_0[rank]} -> "
+                              f"{events_last_event_ts_1[rank]} Fail (Value changed)")
+                else:
+                    result = (f"Rank {rank}: {events_last_event_ts_0[rank]} -> "
+                              f"{events_last_event_ts_1[rank]} Pass (Value unchanged)")
+            else:
+                if events_last_event_ts_0[rank] >= events_last_event_ts_1[rank]:
+                    msg = (f"No increase detected in engine_events_last_events_ts for rank {rank}! "
+                           f"Before = {events_last_event_ts_0[rank]}; "
+                           f"After = {events_last_event_ts_1[rank]}")
+                    errors.append(msg)
+                    result = (f"Rank {rank}: {events_last_event_ts_0[rank]} -> "
+                              f"{events_last_event_ts_1[rank]} Fail (Value decreased/unchanged)")
+                else:
+                    result = (f"Rank {rank}: {events_last_event_ts_0[rank]} -> "
+                              f"{events_last_event_ts_1[rank]} Pass (Value increased)")
+            events_last_event_ts_results.append(result)
+
+    def test_engine_events(self):
+        """Test engine-related events after an engine is restarted.
+
+        1. Check that all ranks have started.
+        2. Gather initial data for the required metrics for all the server nodes.
+        3. Stop a rank.
+        4. Verify the desired rank has stopped successfully.
+        5. Restart the stopped rank after several seconds.
+        6. Verify the desired rank restarted successfully.
+        7. Gather final data for required metrics with the same command as in step 2.
+        8. Compare the before and after values.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=telemetry
+        :avocado: tags=EngineEvents,test_engine_events
+        """
+        # 1. Check that all ranks have started.
+        self.log_step("Check that all ranks have started.")
```

Comment:
Arguably this is already down as part of setUp().  If this didn't pass in setUp() the test would have already failed.

Reply:
I see. I removed this step.